fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -586,15 +589,52 @@ public Response createObject(@QueryParam("checksum") final String checksumDeprec
         }
 
         LOGGER.debug("Finished creating resource with path: {}", newObjectPath);
+        return createUpdateResponse(resource, true);
+    }
 
-        addCacheControlHeaders(servletResponse, result, session);
-
-        final URI location = getUri(result);
+    private Response createUpdateResponse(final FedoraResource resource, final boolean created) throws IOException {
+        addCacheControlHeaders(servletResponse, resource, session);
+        addResourceLinkHeaders(resource, created);
+        if (!created) {
+            return noContent().build();
+        }
 
-        addResourceLinkHeaders(result, true);
+        final URI location = getUri(resource);
+        final Response.ResponseBuilder builder = created(location);
 
-        return created(location).entity(location.toString()).build();
+        if (prefer == null || !prefer.hasReturn()) {
+            if (acceptsPlainText()) {
+                return builder.type(TEXT_PLAIN_TYPE).entity(location.toString()).build();
+            } else {
+                return notAcceptable(mediaTypes(TEXT_PLAIN_TYPE).build()).build();
+            }
+        } else if (prefer.getReturn().getValue().equals("minimal")) {
+            return builder.build();
+        } else {
+            servletResponse.addHeader("Vary", "Accept, Range, Accept-Encoding, Accept-Language");
+            if (prefer != null) {
+                prefer.getReturn().addResponseHeaders(servletResponse);
+            }
+            final RdfNamespacedStream rdfStream = new RdfNamespacedStream(
+                new DefaultRdfStream(translator().reverse().convert(resource()).asNode(), getResourceTriples()),
+                getNamespaces(session()));
+            return builder.entity(rdfStream).build();
+        }
+    }
 
+    private boolean acceptsPlainText() {
+        final List<String> acceptHeaders = headers.getRequestHeader(HttpHeaders.ACCEPT);
```

Comment:
Yes, and I think it'll be better to rework this method a bit to use the MediaType objects isWildcardType()  and isCompatible() methods, which will be more robust than the current impl.